.github/workflows/ci.yml

```diff
@@ -0,0 +1,65 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - 'v*'
+  pull_request:
+    branches:
+      - master
+jobs:
+  Build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+          cache: pip
+          cache-dependency-path: setup.cfg
+      # TODO: look into using python-babel instead of requiring gettext as system dependency
+      - run: sudo apt-get update && sudo apt-get install -y --no-install-recommends gettext
+      - run: pip install -U build
+      - run: python -m build
+      - uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist/
+  Test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            python-version: "3.6"
+          - os: ubuntu-20.04
+            python-version: "3.7"
+          - os: ubuntu-22.04
+            python-version: "3.8"
+          - os: ubuntu-22.04
+            python-version: "3.9"
+          - os: ubuntu-22.04
+            python-version: "3.10"
```

Comment:
Would be good to drop EOL Python versions and use a regular matrix, but that can be a followup 👍

Reply:
Yeah, that was mentioned in #556.